### PR TITLE
[Serve] Reword no Spot Policy in SkyServe

### PR DESCRIPTION
**Pull request by cblmemo**
Previously, when no spot fallback policy was specified, `No spot policy` would be shown on the terminal, which is confusing when users are using spot resources. This PR reword it to `No spot fallback policy` to reduce confusion.

```bash
# previously
$ sky serve up -n http examples/serve/http_server/task.yaml --cloud aws --use-spot
Service from YAML spec: examples/serve/http_server/task.yaml
Service Spec:
Readiness probe method:           GET /health
Readiness initial delay seconds:  20
Readiness probe timeout seconds:  15
Replica autoscaling policy:       Fixed 2 replicas
Spot Policy:                      No spot policy
# this PR
$ sky serve up -n http examples/serve/http_server/task.yaml --cloud aws --use-spot
Service from YAML spec: examples/serve/http_server/task.yaml
Service Spec:
Readiness probe method:           GET /health
Readiness initial delay seconds:  20
Readiness probe timeout seconds:  15
Replica autoscaling policy:       Fixed 2 replicas
Spot Policy:                      No spot fallback policy
```

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
     - [x] Test in the PR description
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

**Review by concretevitamin — APPROVED** [review]
Seems like we want to rename the LHS column from "Spot Policy:" to "Spot Fallback Policy:" too?

**Comment by cblmemo** [comment]
> Seems like we want to rename the LHS column from "Spot Policy:" to "Spot Fallback Policy:" too?

I'm thinking when we later add the spot placer, we might want to reuse this row, which is not "Spot Fallback Policy", so keeping it unchanged now and making the LHS column consistent might reduce confusion. wdyt?

**Comment by concretevitamin** [comment]
Sounds good.
